rfcs/tentative_testdriver_methods.md

```diff
@@ -0,0 +1,32 @@
+# RFC 226: Tentative testdriver methods
+
+## Summary
+Allow tentative methods to be added to testdriver.js that are not yet included in the WebDriver or WebDriver BiDi specifications.
+This should only be allowed where there is consensus between multiple browser vendors and where there is clear progress towards a specification.
+
+## Background
+As part of the [Interop 2025 Accessibility Investigation Area](https://github.com/web-platform-tests/interop-accessibility/issues/148), the intention is to extend browsers and WPT to allow additional accessibility properties to be tested by web platform tests.
+However, there are open questions regarding the shape of the API for exposing these properties, what properties should be exposed, how the tests should be written, etc.
+While the end goal is to extend the WebDriver specification with the required new endpoints, these open questions need to be answered before this is feasible.
+To answer these questions, it would be helpful if browser vendors could collaborate on a tentative but working implementation.
+
+## Details
+1. There must be publicly documented (e.g. as part of an Interop Investigation Area) consensus between at least two browser vendors and intent to work towards a specification.
+2. Any tentative methods added to testdriver.js should be prefixed with `tentative_`.
+3. Any utility methods which call tentative testdriver methods should also be prefixed with `tentative_`.
+4. Any tests directly or indirectly calling tentative testdriver methods must be marked tentative.
+
+## Alternatives considered
+1. Avoid any changes to the WPT repository until the specification is finalised.
+  This makes it very difficult for browser vendors to collaborate and to learn about problems that are much easier to discover and understand "in practice".
+  In contrast, if the specification were developed without a working implementation, there is a much higher risk of fundamental design problems in the specification which are much harder to fix later.
```

Comment:
Perhaps the initial cross-engine prototype of the property bag implementation should include only the properties already exposed by other WebDriver methods, like `computedrole` and `computedlabel`.

Reply:
This would make it easier to standardise on a first pass, for sure. However, that first pass would have no practical benefit, since we can already get those properties using the existing methods. The problem I see with this is that we're less likely to spot problems because the implementation is thus entirely "theoretical": we won't spot the kinds of problems we're likely to see in testing new properties, which is the entire point of this work. In contrast, the current prototype supports a few simple ARIA properties and parent/children, which allows us to test the kinds of things we really want to test with this work and thus spot problems we might hit as we move forward.

Reply:
This should probably be discussed elsewhere, but now that we've ascertained that we don't need to worry about the WebElement-style abstraction for accessibility ids, I think the way in which accessibility ids should be exposed is easy to define: a simple unique string such as a UUID. DOM nodes should be exposed as WebElements just as they are everywhere else. It's just a matter of implementing it. There might be tree structure differences, but that's not relevant to WebDriver exposure; that's a higher level question of which parts of the tree we can standardise (and working around that in tests if need be).